Documentation/Development/KeyCodeConcepts.md

```diff
@@ -0,0 +1,27 @@
+# WheelMUD Key Code Concepts
+
+## Thing
+This is the basic building block for any interactive object object in the game world; whether it be a player, an enemy, a room, a door, a weapon, or even an area itself. Things can have a collection of child objects; a player Thing may have children representing the player's equipment and inventory. (Note that we capitalize Thing when communicating about this code class, to avoid confusion.)
+
+## Behavior
+Any Thing can be adorned with Behaviors. Behaviors modify how a Thing works, giving or taking away capabilities, interactivity, etc. In other frameworks, one might extend the capabilities of objects through inheritance (or worse, hacking a multitude of systems to support some edge cases for new behaviors). However, here, Behaviors extend Things in a way that is highly encapsulated and highly dynamic. For example, we can write an OpensClosesBehaviors once and apply it to a Thing that we intend to be a door. We could apply it to a Thing that we intend to be a treasure chest. It does not matter that we wrote it without considering the case of vehicles; we can apply it to a Thing we intend to be a car, and without any code changes, our car now has a functional, interactive door. A world builder can compose highly interesting objects (without writing any code) by combining and configuring these behaviors.
```

Comment:
It would be great if we could break all of these down into known software pattern descriptions so we can easily describe that as well as educate.